app/controllers/records_controller.rb

```diff
@@ -1,7 +1,11 @@
 class RecordsController < ApplicationController
   include RecordsControllerBehavior
 
-  private
+  protected
+
+  def set_attributes
+    resource.attributes = AttributeURIConverter.new(collect_form_attributes).convert_attributes
```

Comment:
I wanted to convert precisely what's sent to the resource, as I'm not sure what other transformations may be necessary - or if there's any need for those form attributes prior to being converted.  `resource.attributes=` is the last place the controller has any power to change them, so I picked it.

That said, I am not sure I care which method I hit, so if it makes a lot of sense to you to put it in `collect_form_attributes`, I could change it.